### PR TITLE
add e2e bash+goal test for rekeying

### PR DESCRIPTION
**Pull request by algobolson**
### Summary

Add automated test doing account re-keying using `goal` with an account on a small private network.

### Test Plan

This _is_ a test.

### CODE REVIEW

File: test/scripts/e2e_subs/rekey.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+WALLET=$1
+
+gcmd="goal -w ${WALLET}"
+
+ACCOUNT=$(${gcmd} account list|awk '{ print $3 }')
+ACCOUNTB=$(${gcmd} account new|awk '{ print $6 }')
+
+algokey generate > ${TEMPDIR}/rekey
+mnemonic=$(grep 'Private key mnemonic:' < ${TEMPDIR}/rekey | sed 's/Private key mnemonic: //')
+ACCOUNTC=$(grep 'Public key:' < ${TEMPDIR}/rekey | sed 's/Public key: //')
```
Comment (tsachiherman) [inline test/scripts/e2e_subs/rekey.sh:19]: nit:
when using grep, you don't need to "<" into it. just pass the filename.
i.e.
```
grep 'Private key mnemonic:' ${TEMPDIR}/rekey
```

**Comment by algobolson** [comment]
tempfiles are taken care of by test/scripts/e2e_client_runner.py

**Comment by tsachiherman** [comment]
Please add the following:
1. create a linked issue for this PR.
2. ensure that the PR description aligns with the template

I will merge this once the above is done.

**Comment by algobolson** [comment]
implements #1078

**Pull request merged**